### PR TITLE
[3530] Improve slow mode

### LINKED ISSUE #3530: Message cooldown not in sync with the BE in certain cases
### Context & proposal

Steps to reproduce:

1. Find a channel with a cooldown period
2. Send a **message with an attachment**
3. Reply to the message immediately after the cooldown ends

Result:

Message fails to send in both apps with the following printed out in the logs  in UI Components:

```
E/Chat:MessageInputViewModel: Could not send message with cid: messaging:6da0a78b-8880-44cc-867b-9d9a448c80aa. Error message: Status code: 403, with stream code: 60, description: SendMessage failed with error: "user jc is not allowed to send messages on channel messaging:6da0a78b-8880-44cc-867b-9d9a448c80aa under cool down period(10s)". Cause message: null
```

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Locate the issue and sync with the BE team if needed.
- [ ] Fix the issue.

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Closes #3530

Start countdown only after the message with attachments is sent.

### 🧪 Testing

1. Login as Jc or Dmitrii
2. Navigate to "Slow mode 10s" channel
3. Send a message with big attachments
4. Ensure that countdown begins only after the message was sent to the server

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt [hunk 1]
```diff
@@ -686,17 +701,28 @@ public class MessageComposerController(
...
+            enqueue {
+                if (it.isSuccess) {
+                    cooldownTimerJob = scope.launch {
+                        for (timeRemaining in cooldownInterval downTo 0) {
+                            cooldownTimer.value = timeRemaining
+                            delay(OneSecond)
+                        }
+                    }
+                } else {
+                    cooldownTimer.value = 0
```
Comment (JcMinarro) [inline stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt:721]: Should we update the timer to 0 on that case?? :thinking: 
What happens in the case a user is offline, sends 100 messages and then comes back offline?
I think we don't really need to distinguish between success and failure here, whenever the `call` is executed, the timer should start

Comment (filbabic) [reply stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt:724]: Resolved - we reset the timer only if the user is online and the call failed (guidance from BE and mutual agreement).

This way in cases of NNBB and similar things we'll be able to resend/edit properly.

For editing - we don't disable the send UI if we're editing (cooldown doesn't affect edit)

**Review by JcMinarro — APPROVED** [review]
Works as expected

**Review by filbabic — APPROVED** [review]
Cheating but approved on my end!

**Pull request merged**